File: CRUD/C#/Aplicacion con SQL Server V2/Aplicacion/Forms/FormCambiarClave.cs
// Written By Ismael Heredia in the year 2017

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Aplicacion.Data;

namespace aplicacion
{
    public partial class FormCambiarClave : Form
    {

        UsuarioDatos usuarioDatos = new UsuarioDatos();
        Funciones funcion = new Funciones();
        String usuario = "";

        public FormCambiarClave(String usuario_recibido)
        {
            InitializeComponent();
            usuario = usuario_recibido;
        }

        private void FormCambiarContraseña_Load(object sender, EventArgs e)
        {
            txtUsuarioActual.Text = usuario;
        }

        private void btnCambiar_Click(object sender, EventArgs e)
        {

            if (txtUsuarioActual.Text == "" || txtClave.Text == "" || txtNuevaClave.Text == "")
            {
                MessageBox.Show("Faltan datos");
            }
            else
            {
                string usuario = txtUsuarioActual.Text;
                string clave = funcion.md5_encode(txtClave.Text);
                string nueva_clave = funcion.md5_encode(txtNuevaClave.Text);

                int id = usuarioDatos.get_id_by_user(usuario);

                if (usuarioDatos.check_login(usuario,clave))
                {
                    if (usuarioDatos.change_password(id, nueva_clave))
                    {
                        MessageBox.Show("La clave ha sido cambiada , reinicie la aplicación");
                        Application.Exit();
                    }
                    else
                    {
                        MessageBox.Show("Ha ocurrido un error en la base de datos");
                    }
                }
                else
                {
                    MessageBox.Show("Login inválido");
                }
            }
        }
    }
}

File: CRUD/VB.Net/Aplicacion con SQL Server V2/Sistema/Sistema/Forms/FormCambiarClave.vb
' Written By Ismael Heredia in the year 2017

Imports Sistema.Aplicacion.Data

Public Class FormCambiarClave

    Dim usuarioDatos As New UsuarioDatos
    Dim funcion As New Funciones
    Dim usuario As String = ""

    Public Sub New(ByVal usuario_recibido As String)
        InitializeComponent()
        usuario = usuario_recibido
    End Sub

    Private Sub btnCambiarClave_Click(sender As Object, e As EventArgs) Handles btnCambiarClave.Click
        If txtUsuarioActual.Text = "" OrElse txtClave.Text = "" OrElse txtNuevaClave.Text = "" Then
            MessageBox.Show("Faltan datos")
        Else
            Dim usuario As String = txtUsuarioActual.Text
            Dim clave As String = funcion.md5_encode(txtClave.Text)
            Dim nueva_clave As String = funcion.md5_encode(txtNuevaClave.Text)

            Dim id As Integer = usuarioDatos.get_id_by_user(usuario)

            If usuarioDatos.check_login(usuario, clave) Then
                If usuarioDatos.change_password(id, nueva_clave) Then
                    MessageBox.Show("La clave ha sido cambiada , reinicie la aplicación")
                    Application.Exit()
                Else
                    MessageBox.Show("Ha ocurrido un error en la base de datos")
                End If
            Else
                MessageBox.Show("Login inválido")
            End If
        End If
    End Sub

    Private Sub FormCambiarClave_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        txtUsuarioActual.Text = usuario
    End Sub

End Class